.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+name: Release
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  release:
+    types: [ published ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
```

Comment:
please use runs-on: ${{ matrix.config.os }}.
and create builds for arch using https://github.com/marketplace/actions/run-on-architecture